src/ASM/Test/TestASMs3D.C

```diff
@@ -0,0 +1,53 @@
+//==============================================================================
+//!
+//! \file TestASMs3D.C
+//!
+//! \date Feb 14 2018
+//!
+//! \author Arne Morten Kvarving / SINTEF
+//!
+//! \brief Tests for structured 3D spline FE models.
+//!
+//==============================================================================
+
+#include "ASMs3D.h"
+#include "IntegrandBase.h"
+#include "SIM3D.h"
+
+#include "gtest/gtest.h"
+#include <numeric>
+
+
+template<class Dim>
+class DummySIM : public Dim {
+public:
+  class DummyIntegrand : public IntegrandBase {};
+  DummySIM() : Dim(Dim::dimension) { Dim::myProblem = new DummyIntegrand; }
+};
+
+class TestASMs3D : public testing::Test,
+                   public testing::WithParamInterface<int>
+{
+};
+
+
+TEST_P(TestASMs3D, Connect)
+{
+  DummySIM<SIM3D> sim;
+  std::stringstream str;
+  str << "src/ASM/Test/refdata/DomainDecomposition_MPI_3D_4_orient";
+  str << GetParam() << ".xinp";
+  ASSERT_TRUE(sim.read(str.str().c_str()));
+  ASSERT_TRUE(sim.preprocess());
```

Comment:
You don't need to execute the preprocess() method to only test the connection stuff. It suffice to invoke createFEMmodel(). Then the DummySIM class is not needed either (can use SIM3D directly).

Reply:
indeed. i somehow was under the impression the method was protected.